Snowflake.Data.Tests/ConcatenatedStreamTest.cs

```diff
@@ -0,0 +1,159 @@
+﻿/*
+ * Copyright (c) 2023 Snowflake Computing Inc. All rights reserved.
+ */
+
+namespace Snowflake.Data.Tests
+{
+    using NUnit.Framework;
+    using Snowflake.Data.Core;
+    using System;
+    using System.IO;
+    using System.Text;
+
+    [TestFixture]
+    class ConcatenatedStreamTest
+    {
+        Stream concatStream;
+
+        [SetUp]
+        public void BeforeTest()
+        {
+            string data = "12345678";
+            byte[] bytes = Encoding.UTF8.GetBytes(data);
+            Stream stream = new MemoryStream(bytes);
+
+            Stream openBracket = new MemoryStream(Encoding.UTF8.GetBytes(data));
+            Stream closeBracket = new MemoryStream(Encoding.UTF8.GetBytes(data));
+
+            concatStream = new ConcatenatedStream(new Stream[3] { openBracket, stream, closeBracket });
+        }
+
+        [Test]
+        [Ignore("ConcatenatedStreamTest")]
+        public void ConcatenatedStreamTestDone()
+        {
+            // Do nothing;
+        }
+
+        [Test]
+        public void TestCanRead()
+        {
+            Assert.IsTrue(concatStream.CanRead);
+        }
+
+        [Test]
+        public void TestCanSeek()
+        {
+            Assert.IsFalse(concatStream.CanSeek);
+        }
+
+        [Test]
+        public void TestCanWrite()
+        {
+            Assert.IsFalse(concatStream.CanWrite);
+        }
+
+        [Test]
+        public void TestFlush()
+        {
+            try
+            {
+                concatStream.Flush();
+            }
+            catch (Exception ex)
+            {
+                Assert.IsTrue(ex is NotImplementedException);
+            }
+        }
+
+        [Test]
+        public void TestLength()
+        {
+            try
+            {
+                long l = concatStream.Length;
```

Comment:
do we need temporary variable?
we should have always failing assertion at the end of try block - the same problem happens also in the tests below

Reply:
Removed the temporary variable and added `Assert.Fail` at the end of try block for all the relevant tests